snuba/state/rate_limit.py

```diff
@@ -87,15 +87,20 @@ def rate_limit(
     rate_limit_params: RateLimitParameters,
 ) -> Iterator[Optional[RateLimitStats]]:
     """
-    A context manager for rate limiting that allows for limiting based on
-    on a rolling-window per-second rate as well as the number of requests
-    concurrently running.
+    A context manager for rate limiting that allows for limiting based on:
+        * a rolling-window per-second rate
+        * the number of queries concurrently running.
 
-    Uses a single redis sorted set per rate-limiting bucket to track both the
-    concurrency and rate, the score is the query timestamp. Queries are thrown
-    ahead in time when they start so we can count them as concurrent, and
-    thrown back to their start time once they finish so we can count them
-    towards the historical rate.
+    It uses one redis sorted set to keep track of both of these limits
+    The following mapping is kept in redis:
+
+        bucket: SortedSet([(timestamp1, query_id1), (timestamp2, query_id2) ...])
+
+
+    Queries are thrown ahead in time when they start so we can count them
```

Comment:
I think being more explicit here is good, something like:

Queries are stored at a timestamp that is in the future (`time.time()` + `max_query_duration_s`) so that we can count them as concurrent, and then when they finish executing we update the timestamp back to it's start time (decrement timestamp by `max_query_duration_s`), so that we can count them towards the historical rate.